tests/beaker_tests/cisco_interface/interfacelib.rb

```diff
@@ -0,0 +1,66 @@
+###############################################################################
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+# Interface test library
+###############################################################################
+require File.expand_path('../../lib/utilitylib.rb', __FILE__)
+
+# This method overrides the method in utilitylib.rb to set up dependencies
+# for interface tests.
+def test_harness_dependencies(tests, id)
+  logger.info('  * Process test_harness_dependencies (interfacelib)')
+
+  # System-level switchport dependencies
+  if operating_system == 'nexus'
+    config_system_default_switchport?(tests, id)
+    config_system_default_switchport_shutdown?(tests, id)
+  end
+
+  # Misc dependencies
+  config_acl?(tests, id)
+  config_anycast_gateway_mac?(tests, id)
+  config_bridge_domain?(tests, id)
+
+  # Various Cleanups
+  return unless tests[id][:preclean_intf]
+  intf = tests[id][:title_pattern]
+  if intf[/ethernet/i]
+    interface_cleanup(agent, intf)
+  else
+    remove_interface(agent, intf)
+  end
+end
+
+# A global 'anycast gateway mac' is required for some SVI properties
+# TBD: This may not be required for all platforms
+def config_anycast_gateway_mac?(tests, _id)
+  return unless tests.key?(:anycast_gateway_mac)
+  agent = tests[:agent]
+  on(agent, get_vshell_cmd('show runn fabric forwarding'), pty: true)
```

Comment:
Instead of adding another `vshell` dependency, what do you think about creating a `resource_get` method to query the device using the `puppet resource` command?  We have a `resource_set` but a `resource_get` would be a nice compliment.